native/dorgbr.go

```diff
@@ -30,9 +30,17 @@ func (impl Implementation) Dorgbr(vect lapack.DecompUpdate, m, n, k int, a []flo
 		}
 	}
 	if wantq {
-		checkMatrix(m, k, a, lda)
+		if m >= k {
```

Comment:
I think this is clear, but the effective size of the matrix depends on the magnitude of the values.